tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java

```diff
@@ -459,11 +461,14 @@ public void onDrawComplete(IMap map, IFeature feature) {
             MainActivity.this.ePlotMode = PlotModeEnum.IDLE;
             MainActivity.this.oEditorCompleteBtn.hide();
             MainActivity.this.oEditorCancelBtn.hide();
-            MainActivity.this.oFeatureHash.put(feature.getGeoId(), feature);
-            try {
-                MainActivity.this.oRootOverlay.addFeature(feature, true);
-            } catch (EMP_Exception e) {
-                e.printStackTrace();
+            if (!MainActivity.this.oFeatureHash.containsKey(feature.getGeoId())) {
+                // Only add it if it dones not exists. A feature can be placed back into draw mode.
+                MainActivity.this.oFeatureHash.put(feature.getGeoId(), feature);
+                try {
```

Comment:
Should we only add a feature into the feature hash if it is added successfully into `oRootOverlay`?
```java
try {
    MainActivity.this.oRootOverlay.addFeature(feature, true);
    MainActivity.this.oFeatureHash.put(feature.getGeoId(), feature);
} catch (EMP_Exception e) {
```